sql/viewregistry.go

```diff
@@ -0,0 +1,129 @@
+package sql
+
+import (
+	"strings"
+	"sync"
+
+	"gopkg.in/src-d/go-errors.v1"
+)
+
+var (
+	ErrExistingView    = errors.NewKind("the view %s.%s already exists in the registry")
+	ErrNonExistingView = errors.NewKind("the view %s.%s does not exist in the registry")
+)
+
+// View is defined by a Node and has a name.
+type View struct {
+	name       string
+	definition Node
```

Comment:
In the end are we saving the parsed tree instead of the query string? this might cause incompatibilities in the future.

Reply:
Nope, we will be saving the query, but persistence is not implemented yet. It's on the TO-DO list on the PR description. Those things will be implemented in following PRs.

Reply:
Yup, as @erizocosmico said!